iset.mm

```diff
@@ -147216,6 +147339,173 @@ since the target of the homomorphism (operator ` O ` in our model) need
+      cn mpdan eqtr4d ) ACEZAFGZAHIGZAJGZKLMZFGZBFGUJUMREUKUONAJCOPCUMULFRAQSIU
+      LTNULUGEUAUBUCUHUJBUNFAUMBCDUMUDUEUFUI $.
+
+    ${
+      mgptopn.2 $e |- J = ( TopOpen ` R ) $.
+      $( Topology of the multiplication group.  (Contributed by Mario Carneiro,
+         5-Oct-2015.) $)
+      mgptopng $p |- ( R e. V -> J = ( TopOpen ` M ) ) $=
+        ( wcel cts cfv cbs crest co ctopn topnvalg mgptsetg mgpbasg oveq12d cvv
+        eqid cmgp eqtr3d eqtrid wceq fnmgp elex funfvex funfni sylancr eqeltrid
+        wfn syl eqtrd ) ADGZBCHIZCJIZKLZCMIZUMBAMIZUPFUMAHIZAJIZKLURUPUTUSDAUTS
+        ZUSSNUMUSUNUTUOKACDEOUTACDEVAPQUAUBUMCRGUPUQUCUMCATIZREUMTRUJARGVBRGZUD
+        ADUEVCRATATUFUGUHUIUOUNRCUOSUNSNUKUL $.
+    $}
+
+    ${
+      mgpds.2 $e |- B = ( dist ` R ) $.
+      $( Distance function of the multiplication group.  (Contributed by Mario
+         Carneiro, 5-Oct-2015.) $)
+      mgpdsg $p |- ( R e. V -> B = ( dist ` M ) ) $=
+        ( wcel cds cfv cnx cplusg cmulr cop csts co wceq mulrslid slotex dsslid
+        cvv dsndxnplusgndx cslot plusgslid simpri setsslnid eqid mgpvalg fveq2d
+        cn mpdan eqtr4d eqtrid ) BDGZABHIZCHIZFUMUNBJKIZBLIZMNOZHIZUOUMUQTGUNUS
+        PBLDQRDUQUPHTBSUAKUPUBPUPUIGUCUDUEUJUMCURHBUQCDEUQUFUGUHUKUL $.
+    $}
+  $}
+
+
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Ring unit
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c 1r $.
+
+  $( Extend class notation with ring unit. $)
+  cur $a class 1r $.
+
+  $( Define the multiplicative neutral element of a ring.  This definition
+     works by extracting the ` 0g ` element, i.e. the neutral element in a
+     group or monoid, and transferring it to the multiplicative monoid via the
+     ` mulGrp ` function ( ~ df-mgp ).  See also ~ dfur2g , which derives the
+     "traditional" definition as the unique element of a ring which is left-
+     and right-neutral under multiplication.  (Contributed by NM, 27-Aug-2011.)
+     (Revised by Mario Carneiro, 27-Dec-2014.) $)
+  df-ur $a |- 1r = ( 0g o. mulGrp ) $.
+
+  ${
+    ringidval.g $e |- G = ( mulGrp ` R ) $.
+    ringidval.u $e |- .1. = ( 1r ` R ) $.
+    $( The value of the unity element of a ring.  (Contributed by NM,
+       27-Aug-2011.)  (Revised by Mario Carneiro, 27-Dec-2014.) $)
+    ringidvalg $p |- ( R e. V -> .1. = ( 0g ` G ) ) $=
+      ( wcel cur cfv cmgp c0g cvv wceq elex ccom df-ur fveq1i wfn fnmgp fvco2
+      mpan eqtrid syl fveq2i 3eqtr4g ) ADGZAHIZAJIZKIZBCKIUFALGZUGUIMADNUJUGAKJ
+      OZIZUIAHUKPQJLRUJULUIMSLKJATUAUBUCFCUHKEUDUE $.
+  $}
+
+  ${
+    $d e x B $.  $d e x R $.  $d e x V $.
+    dfur2.b $e |- B = ( Base ` R ) $.
+    dfur2.t $e |- .x. = ( .r ` R ) $.
+    dfur2.u $e |- .1. = ( 1r ` R ) $.
+    $( The multiplicative identity is the unique element of the ring that is
+       left- and right-neutral on all elements under multiplication.
```

Comment:
We have several different names for this concept already here: 
* ring unit
* unity element of a ring
* multiplicative neutral element of a ring
* multiplicative identity (of a ring)

We should choose one of them as the name we will use  primarily, and should mention the others in the comment of the definition.

Reply:
Partly because this naming question is already in set.mm, and partly because it affects a lot of places, I am reluctant to try to address it in this pull request.

I also don't thing I have an opinion about one term versus another (or, indeed, whether it would vary depending on things like how explicit we want to be in a particular place or other factors).

Reply:
> Partly because this naming question is already in set.mm, and partly because it affects a lot of places, I am reluctant to try to address it in this pull request.

You are right, this goes far beyond the scope of this PR. Maybe we should create an issue for this.

Reply:
Issue #4512 opened for this.